From: =?utf-8?q?Florian_M=C3=BCllner?= <fmuellner@gnome.org>
Date: Wed, 13 Apr 2022 20:51:55 +0200
Subject: background: Do not queue idle when cancelled

A cancelled cancellable means that the background was destroyed,
so we shouldn't queue an idle or emit the 'loaded' signal anymore.

Bug: https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/5337
Part-of: <https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/2268>
Origin: upstream, 42.1, commit:ef74f922d65d38718b5d8cb13bc43f129947ffc0
---
 js/ui/background.js | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/js/ui/background.js b/js/ui/background.js
index 825fee0..a68f233 100644
--- a/js/ui/background.js
+++ b/js/ui/background.js
@@ -333,6 +333,8 @@ var Background = GObject.registerClass({
             return;
 
         this.isLoaded = true;
+        if (this._cancellable?.is_cancelled())
+            return;
 
         let id = GLib.idle_add(GLib.PRIORITY_DEFAULT, () => {
             this.emit('loaded');
